protocol/storage/miner.go

```diff
@@ -401,7 +400,7 @@ func (sm *Miner) processStorageDeal(proposalCid cid.Cid) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	d := sm.porcelainAPI.DealGet(proposalCid)
+	d := sm.porcelainAPI.DealGet(context.TODO(), proposalCid)
```

Comment:
`Miner#proposalAcceptor` and `Miner#proposalRejector` should be modified to accept a context. 

`Miner#handleMakeDeal` and `Miner#handleQueryDeal` should be modified to _create_ a context, and should pass this context through the call stack.